src/main/java/br/com/alura/comex/controller/ProdutoController.java

```diff
@@ -0,0 +1,80 @@
+package br.com.alura.comex.controller;
+
+import br.com.alura.comex.controller.dto.ProdutoDto;
+import br.com.alura.comex.controller.form.AtualizacaoProdutoForm;
+import br.com.alura.comex.controller.form.ProdutoForm;
+import br.com.alura.comex.model.Produto;
+import br.com.alura.comex.repository.CategoriaRepository;
+import br.com.alura.comex.repository.ProdutoRepository;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.data.domain.Page;
+import org.springframework.data.domain.Pageable;
+import org.springframework.data.domain.Sort;
+import org.springframework.data.web.PageableDefault;
+import org.springframework.http.ResponseEntity;
+import org.springframework.web.bind.annotation.*;
+import org.springframework.web.util.UriComponentsBuilder;
+
+import javax.transaction.Transactional;
+import javax.validation.Valid;
+import java.net.URI;
+import java.util.Optional;
+
+@RestController
+@RequestMapping("/api/produtos")
+public class ProdutoController {
+
+    @Autowired
+    private ProdutoRepository produtoRepository;
+    @Autowired
+    private CategoriaRepository categoriaRepository;
+
+    @GetMapping("/pagina/{numeroPagina}")
+    public Page<ProdutoDto> listaPaginada(@PageableDefault(sort = "nome", direction = Sort.Direction.ASC, size = 5) Pageable pageable) {
+        Page<Produto> produtos = produtoRepository.findAll(pageable);
+        return ProdutoDto.converter(produtos);
+    }
+    @GetMapping("/{id}")
+    public ResponseEntity<ProdutoDto> buscarPorId(@PathVariable Long id) {
+        Optional<Produto> produto = produtoRepository.findById(id);
+        if (produto.isPresent()) {
+            return ResponseEntity.ok(new ProdutoDto(produto.get()));
+        }
+        return ResponseEntity.notFound().build();
```

Comment:
Aqui e nos outros lugares semelhantes daria pra fazer:

```suggestion
        Produto produto = produtoRepository.findById(id).orElseThrow(() -> new ResponseStatusException(HttpStatus.NOT_FOUND));
        return ResponseEntity.ok(new ProdutoDto(produto));
```